marimo/_runtime/executor.py

```diff
@@ -114,28 +162,63 @@ async def execute_cell_async(
         if cell.body is None:
             return None
         assert cell.last_expr is not None
-
-        if _is_coroutine(cell.body):
-            await eval(cell.body, glbls)
-        else:
-            exec(cell.body, glbls)
-
-        if _is_coroutine(cell.last_expr):
-            return await eval(cell.last_expr, glbls)
-        else:
-            return eval(cell.last_expr, glbls)
+        try:
+            if _is_coroutine(cell.body):
+                # fmt: off
+                (
+                    await marimo_output_evaluation
+                    (cell.body, glbls)
+                )
+                # fmt: on
+            else:
+                # fmt: off
+                (
+                    marimo_body_execution
+                    (cell.body, glbls)
+                )
+                # fmt: on
+
+            if _is_coroutine(cell.last_expr):
+                # fmt: off
+                return (
+                    await marimo_output_evaluation
+                    (cell.last_expr, glbls)
+                )
+                # fmt: on
+            else:
+                # fmt: off
+                return (
+                    marimo_output_evaluation
+                    (cell.last_expr, glbls)
+                )
+                # fmt: on
```

Comment:
Instead of formatting our source code to look good in a traceback, would it be possible to just modify the traceback in `cell_runner` to hide marimo internals? This seems difficult to maintain.

Reply:
I think modifying the stack trace, especially with different traces between script mode and notebook mode might introduce more code debt than this.

However, I do agree, that this is a little gimmicky (esp, since lint has to be manually disabled) and not needed.

The current trace of
```python
    return eval(cell.last_expr, glbls)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Is fine, and I think it'd be better to revert to this than introduce stack trace manipulation.
An intermediate is without the formatting

```python
return marimo_output_evaluation(cell.last_expr, glbls)
```

But at that point, the `eval` makes more sense. LMK what you think

Reply:
Okay, makes sense. Reverting to `eval`/`exec` without special formatting sounds good to me.